### PR TITLE
docs: update circuit breaking docs to reflect conn pool changes

### PR DESCRIPTION
**Pull request by snowp**
Signed-off-by: Snow Pettersen <snowp@squareup.com>

After https://github.com/envoyproxy/envoy/pull/4917 pending request circuit breakers is enforced
for h/2. This updates the docs to reflect that. 

*Risk Level*: Low 
*Testing*: n/a 
*Docs Changes*: n/a
*Release Notes*: n/a

### CODE REVIEW

File: docs/root/intro/arch_overview/circuit_breaking.rst [hunk 1]
```diff
@@ -13,10 +13,11 @@ configure and code each application independently. Envoy supports various types
   all hosts in an upstream cluster. In practice this is only applicable to HTTP/1.1 clusters since
   HTTP/2 uses a single connection to each host.
 * **Cluster maximum pending requests**: The maximum number of requests that will be queued while
-  waiting for a ready connection pool connection. In practice this is only applicable to HTTP/1.1
-  clusters since HTTP/2 connection pools never queue requests. HTTP/2 requests are multiplexed
-  immediately. If this circuit breaker overflows the :ref:`upstream_rq_pending_overflow
-  <config_cluster_manager_cluster_stats>` counter for the cluster will increment.
+  waiting for a ready connection pool connection. Since HTTP/2 requests are sent over a single
```
Comment (venilnoronha) [inline docs/root/intro/arch_overview/circuit_breaking.rst]: Did you want to mention the behavior for `HTTP/1.1` also?

File: docs/root/intro/arch_overview/circuit_breaking.rst [hunk 2]
```diff
@@ -13,10 +13,14 @@ configure and code each application independently. Envoy supports various types
...
+  connection, this circuit breaker only comes into play as the initial connection is created,
+  as requests will be multiplexed immediately afterwards. For HTTP/1.1 requests are added to the list
+  of pending requests whenever there aren't enough upstreams connections available to immeditely dispatch
```
Comment (venilnoronha) [inline docs/root/intro/arch_overview/circuit_breaking.rst]: nit: upstreams -> upstream

File: docs/root/intro/arch_overview/circuit_breaking.rst [hunk 3]
```diff
@@ -13,10 +13,14 @@ configure and code each application independently. Envoy supports various types
   all hosts in an upstream cluster. In practice this is only applicable to HTTP/1.1 clusters since
   HTTP/2 uses a single connection to each host.
 * **Cluster maximum pending requests**: The maximum number of requests that will be queued while
-  waiting for a ready connection pool connection. In practice this is only applicable to HTTP/1.1
-  clusters since HTTP/2 connection pools never queue requests. HTTP/2 requests are multiplexed
-  immediately. If this circuit breaker overflows the :ref:`upstream_rq_pending_overflow
-  <config_cluster_manager_cluster_stats>` counter for the cluster will increment.
+  waiting for a ready connection pool connection. Since HTTP/2 requests are sent over a single
+  connection, this circuit breaker only comes into play as the initial connection is created,
+  as requests will be multiplexed immediately afterwards. For HTTP/1.1 requests are added to the list
```
Comment (venilnoronha) [inline docs/root/intro/arch_overview/circuit_breaking.rst]: nit: HTTP/1.1 -> HTTP/1.1,

File: docs/root/intro/arch_overview/circuit_breaking.rst [hunk 4]
```diff
@@ -13,10 +13,14 @@ configure and code each application independently. Envoy supports various types
...
   HTTP/2 uses a single connection to each host.
 * **Cluster maximum pending requests**: The maximum number of requests that will be queued while
-  waiting for a ready connection pool connection. In practice this is only applicable to HTTP/1.1
-  clusters since HTTP/2 connection pools never queue requests. HTTP/2 requests are multiplexed
-  immediately. If this circuit breaker overflows the :ref:`upstream_rq_pending_overflow
-  <config_cluster_manager_cluster_stats>` counter for the cluster will increment.
+  waiting for a ready connection pool connection. Since HTTP/2 requests are sent over a single
+  connection, this circuit breaker only comes into play as the initial connection is created,
+  as requests will be multiplexed immediately afterwards. For HTTP/1.1, requests are added to the list
+  of pending requests whenever there aren't enough upstream connections available to immeditely dispatch
```
Comment (mattklein123) [inline docs/root/intro/arch_overview/circuit_breaking.rst]: typo immeditely

**Pull request merged**